src/Oceanostics.jl

```diff
@@ -42,7 +42,8 @@ using Oceananigans.TurbulenceClosures: νᶜᶜᶜ, calc_nonlinear_κᶜᶜᶜ
 @inline _νᶜᶜᶜ(i, j, k, grid, closure_tuple::Tuple{}, K::Tuple{}, clock) = zero(eltype(grid))
 @inline _calc_nonlinear_κᶜᶜᶜ(i, j, k, grid, closure_tuple::Tuple{}, args...) = zero(eltype(grid))
 
-# "Inner-outer" form makes the compiler "unroll" the loop over a tuple:
+# Unroll the loop over a tuple. This is necessary because no arrays can be dynamically
```

Comment:
Sorry to nitpick but I think the description is a little misleading. It seems to imply that there are only two ways to compute a sum: 1) dynamically allocating arrays and then summing over the arrays or 2) this method.

But neither are true...

The only point is that whatever method we use it has to be statically inferrable by the compiler. Who knows how many valid methods there are (it doesn't matter), we just found one that works so... good!

It rarely makes sense to allocate within a kernel, either on CPU or GPU. I think it's better to avoid mentioning all the ways this _can't_ be done...